torch/nn/parallel/distributed.py

```diff
@@ -646,8 +644,8 @@ def __init__(
             )
 
         # Check that a module does not have Uninitialized parameters
-        for param in module.parameters():
-            if isinstance(param, torch.nn.parameter.UninitializedParameter):
+        for name, param in module.named_parameters():
```

Comment:
shall we directly check `self._module_parameters` ?

Reply:
I think this works